Blog post: Make running programs through AiiDA a piece of cake

sphuber opened the pull request.

sphuber commented:
@GeigerJ2 is there actually a way to get the post built into the actual docs in order to preview it on the site? Or do we just have to go off the markdown rendered by Github in the rich diff view?

GeigerJ2 commented:
Sure, there ya go:
https://aiida-website--75.org.readthedocs.build/news/posts/2024-11-01-aiida-shell.html
It doesn't show up under the Posts tab of the RTD that is built through the readthedocs Action (probably because the date is in the future), but it's possible to find it via the search.
Title checks out! I'll add my review tomorrow. Thanks!

GeigerJ2 commented on docs/news/posts/2024-11-01-aiida-shell.md:
```diff
@@ -0,0 +1,314 @@
+
+To make the usefulness and power of `aiida-shell` more concrete, let's now go through a real life use case.
+In the following, we will compute the electronic band structure of gallium arsenide using Quantum ESPRESSO.
+[Quantum ESPRESSO](https://www.quantum-espresso.org/) is a free and open-source an integrated suite of Open-Source computer codes for electronic-structure calculations and materials modeling at the nanoscale.
```
```suggestion
[Quantum ESPRESSO](https://www.quantum-espresso.org/) is a free and Open-Source integrated suite of computer codes for electronic-structure calculations and materials modeling at the nanoscale.
```

GeigerJ2 commented on docs/news/posts/2024-11-01-aiida-shell.md:
````diff
@@ -0,0 +1,314 @@
+To make them available to the calculations, we download and store them in a `FolderData` node:
+
+```python
+#!/usr/bin/env runaiida
````
Or, alternatively to my suggestion of highlighting `verdi run` above, we could also add this shebang.

sphuber replied:
Yeah, in the example of the aiida-shell docs, I have provided the entire file as script and so the shebang guarantees loading the profile. I will leave this as is, and have added a comment above to use `verdi run` or `verdi shell`.

GeigerJ2 commented on docs/news/posts/2024-11-01-aiida-shell.md:
````diff
@@ -0,0 +1,314 @@
+```
+
+The next step is to launch the the self-consistent calculation.
+Below we define the input script:
````
```suggestion
Below we define the input script using the Quantum ESPRESSO input format:
```

GeigerJ2 commented on docs/news/posts/2024-11-01-aiida-shell.md:
```diff
@@ -0,0 +1,314 @@
+In the snippet above, the [`date` program](https://www.man7.org/linux/man-pages/man1/date.1.html) is used as an example, which is a simple Linux utility that prints the current date.
+The program is simply passed as a string to the `launch_shell_job` function.
+Under the hood, `aiida-shell` will find the location of this executable and write the necessary plugins on-the-fly.
+It then runs the code as one would normally do with AiiDA, and the outputs are parsed.
```
```suggestion
It then runs the code (on the `localhost` Computer) as one would normally do with AiiDA, and the outputs are parsed.
```

GeigerJ2 commented on docs/news/posts/2024-11-01-aiida-shell.md:
````diff
@@ -0,0 +1,314 @@
+)
+```
+
+With the bands computed, we need to extract them from the `output.save` directory in a format that allows them to be plotted.
````
```suggestion
In this manner, `aiida-shell` easily allows chaining multiple simulations together, keeping provenance of the full workflow. With the bands computed, we need to extract them from the `output.save` directory in a format that allows them to be plotted.
```

GeigerJ2 reviewed (approved):
Very nice work, @sphuber. Added a few minor comments, but feel free to ignore!

Not sure why `linkcheck` is failing again, but it has done so for every single blog post PR for unknown reasons on links that worked previously, so I think we can just ignore it.

GeigerJ2 commented:
One note, maybe we can provide the full script as a downloadable Python file at the beginning of the "Real life scientific use case" section?

sphuber commented on docs/news/posts/2024-11-01-aiida-shell.md:
```diff
@@ -0,0 +1,314 @@
+
+In the snippet above, the [`date` program](https://www.man7.org/linux/man-pages/man1/date.1.html) is used as an example, which is a simple Linux utility that prints the current date.
+The program is simply passed as a string to the `launch_shell_job` function.
+Under the hood, `aiida-shell` will find the location of this executable and write the necessary plugins on-the-fly.
```
It is not *strictly* correct, but it uses the `ShellJob` calcjob plugin and `ShellParser` parser plugin to run the program. I decided to keep the language such that hopefully it is understandable for new AiiDA users that don't really know what a plugin is anyway.

GeigerJ2 replied:
Ah, I see. Yeah, I find the use of the word _plugin_ for these entities, as well as transport and scheduler, a bit confusing, as _plugin_ for me are still things like `aiida-quantumespresso`. But I get that this is the nomenclature used.

sphuber commented on docs/news/posts/2024-11-01-aiida-shell.md:
````diff
@@ -0,0 +1,314 @@
+verdi presto
+```
+
+If there are no error messages, you should now be good to go to follow the rest of the examples.
````
```suggestion
If there are no error messages, you should now be good to go to follow the rest of the examples.
You can either add the snippets to a file, e.g. `script.py`, and run it with `verdi run script.py`.
Or you can open an interactive shell with `verdi shell`, and run the commands interactively.
```

sphuber commented:
Thanks for the review @GeigerJ2 . I have addressed the comments. For me this is fine to publish. I will leave the final merge to you should you still want to do anything. Not sure if anything else is needed.

GeigerJ2 commented:
Just a ping to @sphuber that I just squash-merged this. Thanks!
